src/Illuminate/Support/HigherOrderCollectionProxy.php

```diff
@@ -54,7 +54,15 @@ public function __get($key)
     public function __call($method, $parameters)
     {
         return $this->collection->{$this->method}(function ($value) use ($method, $parameters) {
-            return $value->{$method}(...$parameters);
+            if (is_object($value)) {
+                return $value->{$method}(...$parameters);
+            }
+
+            if ($method == 'empty') {
```

Comment:
> Exactly, you cannot call empty as a dynamic / variable function (PHP limitation)

Hmm, seems odd that PHP wouldn't allow that function to be called in that way, maybe even a bug? Is there any documentation on this, and are there any other functions we should be aware of that have similar limitations?

> with echo

That's actually not implemented as a function though, it's just a prefix operator.

> I don't think you need array_merge either, the test using substr(3, -3) worked as expected

Yeh, you're right. I was just thinking about unpacking an empty set of parameters, but it seems that works just fine.